### PR TITLE
Preparatory work for adding a custom LiveServerTestCase for process tests that use Resolwe API

### PR DESCRIPTION
**Pull request by tjanez**
Contains:
- Upgrade to Sphinx 1.5.1+
- Remove `RESOLWE_API_HOST` setting from test project's settings since it is not really used anywhere
- Expose `check_docker()` test utility function 
- Add `check_installed()` test utility function
- Add support for configuring the network mode of Docker executor

I think it will be easier to review this commit by commit.

### CODE REVIEW

File: resolwe/flow/executors/docker.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ def start(self):
             data_id = 'test_{}'.format(random.randint(1000, 9999))
         command_args['container_name'] = '--name=resolwe_{}'.format(data_id)
 
+        # configure Docker network mode for the container (if specified)
```
Comment (mstajdohar) [inline resolwe/flow/executors/docker.py]: Start comments with a capital letter. Also write what is the default networking mode that is used here. From PEP8:
> Comments should be complete sentences. If a comment is a phrase or sentence, its first word should be capitalized, unless it is an identifier that begins with a lower case letter (never alter the case of identifiers!).

File: resolwe/flow/tests/test_access_api.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+@unittest.skipUnless(*check_installed('curl'))
+class AccessAPILocalExecutorTestCase(LiveServerTestCase):
+
+    def setUp(self):
+        user_model = get_user_model()
+        self.contributor = user_model.objects.create_user('test_user', 'test_pwd')
+
+    def tearDown(self):
+        for data in Data.objects.all():
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_access_api.py]: We do the same in many tests. Consider writing a helper function.

File: resolwe/flow/tests/test_access_api.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            data_dir = os.path.join(settings.FLOW_EXECUTOR['DATA_DIR'], str(data.id))
+            shutil.rmtree(data_dir, ignore_errors=True)
+
+    def test_access_api_from_local_executor_process(self):  # pylint: disable=invalid-name
+        """Test if a process running via local executor can access API."""
+        # create a test collection that we'll access via a test process
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_access_api.py]: Capitalize comments.

File: resolwe/flow/tests/test_access_api.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        data.refresh_from_db()
+
+        # check if information about the test collection obtained via API is
+        # correct
+        collection_list = Storage.objects.get(data=data).json
+        self.assertEqual(len(collection_list), 1)
+        self.assertEqual(collection_list[0]['slug'], collection_name)
+
+
+flow_executor_settings = copy.copy(  # pylint: disable=invalid-name
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_access_api.py]: Consider capitalizing this constant.

File: resolwe/flow/tests/test_access_api.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+@unittest.skipUnless(*check_docker())
+@override_settings(FLOW_EXECUTOR=flow_executor_settings)
+class AccessAPIDockerExecutorTestCase(LiveServerTestCase):
+
+    def setUp(self):
+        user_model = get_user_model()
+        self.contributor = user_model.objects.create_user('test_user', 'test_pwd')
+
+    def tearDown(self):
+        for data in Data.objects.all():
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_access_api.py]: We do the same in many tests. Consider writing a helper function.

File: resolwe/flow/tests/test_access_api.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            data_dir = os.path.join(settings.FLOW_EXECUTOR['DATA_DIR'], str(data.id))
+            shutil.rmtree(data_dir, ignore_errors=True)
+
+    def test_access_api_from_docker_executor_process(self):   # pylint: disable=invalid-name
+        """Test if a process running via Docker executor can access API."""
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_access_api.py]: This test is practically the same as the above. How about abstracting the duplicated code? Maybe write a class method assertAccessAPI.

File: resolwe/flow/utils/test.py [hunk 7]
```diff
@@ -13,26 +13,73 @@
...
+from django.test import LiveServerTestCase as DjangoLiveServerTestCase
 from django.utils.crypto import get_random_string
 from django.utils.text import slugify
 
 from resolwe.flow.models import (Data, dict_dot, iterate_fields, iterate_schema, Collection,
                                  DescriptorSchema, Process, Storage)
 from resolwe.flow.managers import manager
 
+if six.PY2:
+    # monkey-patch shutil package with which function (available in Python 3.3+)
```
Comment (mstajdohar) [inline resolwe/flow/utils/test.py]: Capitalize comments.

File: resolwe/flow/utils/test.py [hunk 8]
```diff
@@ -13,26 +13,73 @@
...
+        except OSError:
+            available, reason = False, "Docker command '{}' not found".format(command)
+        except subprocess.CalledProcessError:
+            available, reason = (False, "Docker command '{}' returned non-zero "
+                                        "exit status".format(info_command))
+    return available, reason
+
+
 # override all FLOW_EXECUTOR settings that are specified in FLOW_EXECUTOR['TEST']
 flow_executor_settings = copy.copy(getattr(settings, 'FLOW_EXECUTOR', {}))  # pylint: disable=invalid-name
```
Comment (mstajdohar) [inline resolwe/flow/utils/test.py]: Consider capitalizing the constant.

**Comment by mstajdohar** [comment]
Also, review why tests fail.

File: tests/settings.py [hunk 9]
```diff
@@ -10,8 +10,6 @@
 
 SECRET_KEY = 'secret'
 
-RESOLWE_API_HOST = 'localhost'
```
Comment (tjanez) [inline tests/settings.py]: I think it doesn't really scale to have the test project have all possible setting values set and serve as a reference. I think we would need proper configuration documentation, hence I opened #189.

File: resolwe/flow/executors/docker.py (hunk 1 above)
Comment (tjanez) [inline resolwe/flow/executors/docker.py]: Fixed.

File: resolwe/flow/utils/test.py (hunk 7 above)
Comment (tjanez) [inline resolwe/flow/utils/test.py]: Fixed.

File: resolwe/flow/utils/test.py (hunk 8 above)
Comment (tjanez) [inline resolwe/flow/utils/test.py]: Done.

**Comment by tjanez** [comment]
@mstajdohar, I've fixed preparatory commits with your suggestions and I'll leave the last commit out to accept this PR as soon as possible.

**Comment by tjanez** [comment]
The coverage part is a bit unfortunate. The [`check_installed()`](https://codecov.io/gh/genialis/resolwe/compare/92593b50531448072da4c7d73ddbf5af7306aa74...d895f8763776f9e400484e2ba357507f7063c475#7265736F6C77652F666C6F772F7574696C732F746573742E7079-43) function will be covered after I re-add the test cases that will use Resolwe API.

**Pull request merged**